### DISCUSSION TITLE
6-1 Cannot Connect New Data Store

### DISCUSSION THREAD

**Discussion by sidneyh-2** [ua-gist604b-f24/syllabus#26] [General]
Is anyone else having issues creating the data store in GeoServer? I cannot connect in 6-1 or 6-2 and I am following the directions. I get the following message but do not understand it. 
<img width="759" alt="Screenshot 2024-11-20 at 3 15 08 PM" src="https://github.com/user-attachments/assets/35f43262-5014-4171-9329-8f5c16190ab5">

**Comment by andreasrf** [comment]
I've actually been having this same issue! I was able to connect to postgres on my terminal using `psql -h localhost -p 5432 -U postgres -d hawaii`, which told me that the issue was with docker. I went into the `docker-compose.yml` file and checked the `hostname` for that, which was `postgis` on mine. I used that in the `host` for the connection parameters on Geoserver and that did the trick.

Essentially, docker and postgres were running on 2 different networks, with postgres inside of docker. So when Geoserver tried to connect to postgres it couldn't get to postgres because postgres is living inside docker. What I ended up doing was basically telling it to connect to docker and *then* access postgres inside it.

These are the Stack Exchange pages I used which pointed me in the right direction:
> https://stackoverflow.com/questions/78040720/goeserver-adding-store-unable-to-obtain-connection-cannot-create-poolableconn
> https://stackoverflow.com/questions/20825734/postgresql-connection-refused-check-that-the-hostname-and-port-are-correct-an

I hope this works for you!! 
Andrea

**Reply by andreasrf** [reply]
Pffft, I kept reading on the assignment and just saw that Aaryn said the `hostname` is `postgis`. Guess now we know why! XD 
![image](https://github.com/user-attachments/assets/9403ca37-f687-421e-b53d-397bad8abea6)

**Reply by sidneyh-2** [reply]
THANK YOU!!! I did see that postgis was the hostname but stupidly I got myself confused and put that in the user spot instead.

**Reply by sidneyh-2** [reply]
Did you also get connected in 6-2? I am still having this issue despite changing the host to 'postgis'.

**Reply by andreasrf** [reply]
I haven't gotten to 6-2 just yet but when I do I'll let you know!
